src/main/java/io/github/haykam821/withersweeper/game/phase/WithersweeperActivePhase.java

```diff
@@ -208,6 +212,31 @@ private ActionResult useBlock(ServerPlayerEntity uncoverer, Hand hand, BlockHitR
 		return result;
 	}
 
+	private void autoReveal(ServerPlayerEntity uncoverer, BlockPos clickedBlockPos, Field clickedField) {
+		if (clickedField.getBlockState() == Blocks.WHITE_WOOL.getDefaultState()) {
+			List<BlockPos> blockPosToCheck = new ArrayList<>();
+			List<BlockPos> checkedBlockPos = new ArrayList<>();
+			blockPosToCheck.add(clickedBlockPos);
+			while (!blockPosToCheck.isEmpty()) {
+				for (int x = -1; x < 2; x++) {
+					for (int z = -1; z < 2; z++) {
+						if (checkedBlockPos.contains(blockPosToCheck.get(0))) continue;
+						BlockPos blockPos = new BlockPos(blockPosToCheck.get(0).getX() + x, blockPosToCheck.get(0).getY(), blockPosToCheck.get(0).getZ() + z);
+						Field field = this.board.getField(blockPos.getX(), blockPos.getZ());
+						if (field == null) {
+							checkedBlockPos.add(blockPos);
+							continue;
+						}
+						if (field.getVisibility() == FieldVisibility.COVERED) this.modifyField(uncoverer, blockPos, field);
+						if (field.getBlockState() == Blocks.WHITE_WOOL.getDefaultState()) blockPosToCheck.add(blockPos);
+					}
+				}
+				checkedBlockPos.add(blockPosToCheck.get(0));
+				blockPosToCheck.remove(blockPosToCheck.get(0));
```

Comment:
Just specify the index here:

```suggestion
				blockPosToCheck.remove(0);
```